app/slashcommands/command_remote.go

```diff
@@ -0,0 +1,212 @@
+	}
+
+	margs := parseNamedArgs(args.Command)
+	action, ok := margs[ActionKey]
+	if !ok {
+		return responsef("Missing command. " + AvailableRemoteActions)
+	}
+
+	switch action {
+	case "add":
+		return rp.doAdd(a, args, margs)
+	case "remove":
+		return rp.doRemove(a, args, margs)
+	case "status":
+		return rp.doStatus(a, args, margs)
+	}
+
+	return responsef("Unknown action `%s`", action)
+}
+
+func (rp *RemoteProvider) GetAutoCompleteListItems(a *app.App, commandArgs *model.CommandArgs, arg *model.AutocompleteArg, parsed, toBeParsed string) ([]model.AutocompleteListItem, error) {
+	if !a.HasPermissionTo(commandArgs.UserId, model.PERMISSION_MANAGE_SHARED_CHANNELS) {
+		return nil, errors.New("You require `manage_shared_channels` permission to manage remote clusters.")
+	}
+
+	if arg.Name == "remoteId" && strings.Contains(parsed, " remove ") {
+		return getRemoteClusterAutocompleteListItems(a, true)
+	}
+
+	return nil, fmt.Errorf("`%s` not a dynamic argument", arg.Name)
+}
+
+func (rp *RemoteProvider) doAdd(a *app.App, args *model.CommandArgs, margs map[string]string) *model.CommandResponse {
+	name := margs["name"]
+	if name == "" {
+		return responsef("Missing or empty `name`")
+	}
+
+	host := margs["host"]
+	if host == "" {
+		return responsef("Missing or empty `host`")
+	}
+
+	port := margs["port"]
+	if port == "" {
+		port = "8065"
+	}
+	iport, err := strconv.ParseInt(port, 10, 32)
+	if err != nil {
+		return responsef("Invalid `port`: %v", err)
+	}
+
+	rc := &model.RemoteCluster{
+		ClusterName: name,
+		Hostname:    host,
+		Port:        int32(iport),
+	}
+
+	rcSaved, err := a.AddRemoteCluster(rc)
+	if err != nil {
+		responsef("Could not add remote cluster: %v", err)
+	}
+	return responsef("##### Remote cluster added.\nName: %s\nHost: %s\nPort: %d\nToken: %s", rcSaved.ClusterName, rc.Hostname, rc.Port, rcSaved.Token)
+}
+
+func (rp *RemoteProvider) doRemove(a *app.App, args *model.CommandArgs, margs map[string]string) *model.CommandResponse {
+	id, ok := margs["remoteId"]
+	if !ok {
+		return responsef("Missing `remoteId`")
+	}
+
+	deleted, err := a.DeleteRemoteCluster(id)
+	if err != nil {
+		responsef("Could not remove remote cluster: %v", err)
+	}
+
+	result := "removed"
+	if !deleted {
+		result = "**NOT FOUND**"
+	}
+	return responsef("##### Remote cluster %s %s.", id, result)
+}
+
+func (rp *RemoteProvider) doStatus(a *app.App, args *model.CommandArgs, margs map[string]string) *model.CommandResponse {
+	list, err := a.GetAllRemoteClusters(true)
+	if err != nil {
+		responsef("Could not fetch remote clusters: %v", err)
+	}
+
+	if len(list) == 0 {
+		return responsef("** No remote clusters found. **")
+	}
+
+	var sb strings.Builder
+	fmt.Fprintf(&sb, "| Name | Host | Token | Id   | Online |\n")
+	fmt.Fprintf(&sb, "| ---- | ---- | ----  | ---- | ----   |\n")
+
+	for _, rc := range list {
+		online := ":white_check_mark:"
+		if !isOnline(rc.LastPingAt) {
+			online = ":skull_and_crossbones:"
```

Comment:
💯 ☠️ 

nit: We can also use 🟢 🔴 notation.

Reply:
I'm going to have a bit of fun with this one until a PM changes it.  Thanks.